e2e_tests/tests/conftest.py

```diff
@@ -0,0 +1,125 @@
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from selenium import webdriver
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.chrome.options import Options as ChromeOptions
+from selenium.webdriver.common.by import By
+from selenium.webdriver.firefox.options import Options as FirefoxOptions
+from selenium.webdriver.remote.webelement import WebElement
+
+LOG = logging.getLogger(__name__)
+
+
+def get_element_by_test_id(driver: webdriver, test_id: str) -> WebElement:
+    """Get HTML DOM element with a given test id.
+
+    Args:
+        driver (webdriver): Selenium web driver
+        test_id (str): id for DOM element
+
+    Raises:
+        NoSuchElementException: raised if no element was found.
+
+    Returns:
+        WebElement: Returns the element
+    """
+    try:
+        element: WebElement = driver.find_element(
+            By.XPATH, f"//*[@data-test-id='{test_id}']"
+        )
+    except NoSuchElementException:
+        raise NoSuchElementException(f"No element with the test id: {test_id}")
+    return element
+
+
+def get_bootstrap_alert(driver, severity: str = "all") -> WebElement | None:
+    """Get bootstrap alert."""
+    query_class_names = "alert"
+    if not severity == "all":
+        query_class_names = f"{query_class_names} alert-{severity}"
+
+    # look for alert DOM element
+    try:
+        element: WebElement = driver.find_element(By.CLASS_NAME, query_class_names)
+    except NoSuchElementException:
+        LOG.debug("No alert was found!")
+        return None
+    return element
+
+
+@pytest.fixture(scope="session")
+def remote_driver(config) -> webdriver:
+    """Setup remote driver."""
+    browsers = {
+        "chrome": ChromeOptions,
+        "firefox": FirefoxOptions,
+    }
```

Comment:
i guess this sets up some sort of profile for selenium mimicking specific browsers.

if so, is microsoft edge something that can be added? dunno how tech-savvy the the end-users of bonsai are/will be but i guess most of them will use edge.

Reply:
Yes we can run edge as well. I currently only run chrome beacause that was the webdriver I had previously set up. But in the future I wanted to setup so all tests are run with chrome, edge and firefox but this requires the setup of a selenium grid and was a little bit out of scope for this.

Reply:
gotcha